dask/array/rechunk.py

```diff
@@ -214,7 +213,16 @@ def rechunk(x, chunks, threshold=None,
     >>> y = x.rechunk({0: -1, 1: 'auto'}, block_size_limit=1e8)
     """
     if isinstance(chunks, dict):
-        chunks = dict(chunks)
+        new = {}
+        for k, v in chunks.items():
+            if not isinstance(k, int) or k >= x.ndim or k <= -x.ndim:
```

Comment:
`k=-x.ndim` is valid, e.g., `{-2: 1}` for a 2D array. I usually use `not -ndim <= axis < ndim` for this check.

Reply:
e.g., see https://github.com/pydata/xarray/blob/1688a59803786a9d88eeb43aa4c935f7052d6a80/xarray/core/nputils.py#L9